providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/process/targeting/Operator.java

```diff
@@ -53,34 +59,38 @@ public Object apply(final String flagKey, final String targetingRule, final Eval
     @Getter
     static class FlagProperties {
         private final String flagKey;
+        private final String timestamp;
         private final String targetingKey;
 
         FlagProperties(Object from) {
             if (from instanceof Map) {
                 Map<?, ?> dataMap = (Map<?, ?>) from;
 
-                final Object flagKey = Optional.ofNullable(dataMap.get(FLAGD_PROPS_KEY))
-                    .filter(flagdProps -> flagdProps instanceof Map)
-                    .map(flagdProps -> ((Map<?, ?>)flagdProps).get(FLAG_KEY))
-                    .orElse(null);
-
-                if (flagKey instanceof String) {
-                    this.flagKey = (String) flagKey;
-                } else {
-                    this.flagKey = null;
-                }
+                this.flagKey = extractSubPropertyFromFlagd(dataMap, FLAG_KEY);
+                this.timestamp = extractSubPropertyFromFlagd(dataMap, TIME_STAMP);
 
                 final Object targetKey = dataMap.get(TARGET_KEY);
-
+                
                 if (targetKey instanceof String) {
                     targetingKey = (String) targetKey;
                 } else {
                     targetingKey = null;
                 }
+                
             } else {
                 flagKey = null;
+                timestamp = null;
                 targetingKey = null;
             }
         }
+
+        private String extractSubPropertyFromFlagd(Map<?, ?> dataMap, String flagdKey) {
+            return Optional.ofNullable(dataMap.get(FLAGD_PROPS_KEY))
+                    .filter(flagdProps -> flagdProps instanceof Map)
+                    .map(flagdProps -> ((Map<?, ?>) flagdProps).get(flagdKey))
+                    .filter(value -> value instanceof String)
+                    .map(String::valueOf)
+                    .orElse(null);
+        }
```

Comment:
```suggestion
        private String extractSubPropertyFromFlagd(Map<?, ?> dataMap, String propertyName) {
            return Optional.ofNullable(dataMap.get(FLAGD_PROPS_KEY))
                    .filter(flagdProps -> flagdProps instanceof Map)
                    .map(flagdProps -> ((Map<?, ?>) flagdProps).get(propertyName))
                    .filter(value -> value instanceof String)
                    .map(String::valueOf)
                    .orElse(null);
        }
```

This is not a flag key.